### PR TITLE
Add GitHub actions for gem release and linting

### PR DESCRIPTION
**Pull request by bastian-src**
Use the official Foreman GitHub action to release new gems.

Additionally: Add yamllint action.

### CODE REVIEW

**Review by dosas — APPROVED** [review]

File: .github/workflows/release_gem.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  push:
+    # Pattern matched against refs/tags
+    tags: ['**']
+
+jobs:
+  release:
+    name: Release gem
+    uses: theforeman/actions/.github/workflows/release-gem.yml@v0
+    with:
+      allowed_owner: ATIX-AG
```
Comment (dosas) [inline .github/workflows/release_gem.yaml:14]: Should be 'theforeman'. See also https://github.com/theforeman/foreman_ansible/blob/master/.github/workflows/release_gem.yml#L5

Comment (bastian-src) [reply .github/workflows/release_gem.yaml]: Done, thanks!

File: .github/workflows/yamllint.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+name: Yaml Lint
+# yamllint disable-line rule:truthy
+on: [push]
+jobs:
+  lintAllTheThings:
```
[the remark this answers is not in the document]
Comment (bastian-src) [reply .github/workflows/yamllint.yaml]: basically linting the .github files (and rubocop.yml)

File: .github/workflows/yamllint.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+name: Yaml Lint
+# yamllint disable-line rule:truthy
+on: [push]
```
[the remark this answers is not in the document]
Comment (bastian-src) [reply .github/workflows/yamllint.yaml]: It doesn't. Fixed.

File: .github/workflows/release_gem.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    tags: ['**']
+
+jobs:
+  release:
+    name: Release gem
+    uses: theforeman/actions/.github/workflows/release-gem.yml@v0
+    with:
+      allowed_owner: theforeman
+    secrets:
+      api_key: ${{ secrets.RUBYGEM_API_KEY }}
```
[the remark this answers is not in the document]
Comment (evgeni) [reply .github/workflows/release_gem.yaml:17]: I can create you one, sure.

[the remark this answers is not in the document]
Comment (evgeni) [reply .github/workflows/release_gem.yaml:17]: done

**Pull request merged**